conformance/lang/expressions/regexp/regexp_semantics.balt

```diff
@@ -0,0 +1,719 @@
+
+    function init(string:RegExp reg) {
+        self.reg = reg;
+    }
+
+    function getRegExp() returns string:RegExp {
+        return self.reg;
+    }
+
+    function setRegExp(string:RegExp reg) {
+        self.reg = reg;
+    }
+}
+
+function init() {
+    A classA = new(re `([A-D]+|[H-Z]*)|C{2,9}`);
+    io:println(classA.reg == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(classA.getRegExp() == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    classA.setRegExp(re `[A-Z]`);
+    io:println(classA.reg == re `[A-Z]`); // @output true
+    io:println(classA.getRegExp() == re `[A-Z]`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression with object type.
+Labels: RegExp, object-constructor-expr, var
+
+function init() {
+    var obj = object {
+        string:RegExp reg = re `([A-D]+|[H-Z]*)|C{2,9}`;
+        
+        function getRegExp() returns string:RegExp {
+            return self.reg;
+        }
+        
+        function setRegExp(string:RegExp reg) {
+            self.reg = reg;
+        }
+    };
+
+    io:println(obj.reg == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(obj.getRegExp() == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    obj.setRegExp(re `[A-Z]`);
+    io:println(obj.reg == re `[A-Z]`); // @output true
+    io:println(obj.getRegExp() == re `[A-Z]`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression with type cast expression.
+Labels: RegExp, type-cast-expr
+
+type userType string:RegExp|json|xml;
+
+function init() {
+    anydata x1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    var x2 = <string:RegExp> x1;
+    io:println(x2 is string:RegExp); // @output true
+    io:println(x2 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    any x3 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    var x4 = <string:RegExp> x3;
+    io:println(x4 is string:RegExp); // @output true
+    io:println(x4 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    userType x5 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    var x6 = <string:RegExp> x5;
+    io:println(x6 is string:RegExp); // @output true
+    io:println(x6 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression with stream.
+Labels: RegExp, module-class-defn, module-type-defn, stream
+
+class RegExpGenerator {
+    int i = 0;
+    public isolated function next() returns record {| string:RegExp value; |}|error? {
+        self.i += 1;
+        return { value: re `${self.i}` };
+    }
+}
+
+type ResultValue record {|
+    string:RegExp value;
+|};
+
+function init() {
+    RegExpGenerator regGen = new();
+
+    stream<string:RegExp, error?> regExpStream = new(regGen);
+
+    var regExp = regExpStream.next();
+    
+    if (regExp is ResultValue) {
+        io:println(regExp.value.toString()); // @output 1
+    }
+}
+
+Test-Case: output
+Description: Test 1 for regular expression with query expression.
+Labels: RegExp, from-clause, query-expr, select-clause
```

Comment:
missing `array-type`, `list-constructor-expr`. 
We don't have separate labels for the clauses in query expr